pkg/store/sqlite/search.go

```diff
@@ -164,3 +172,105 @@ func NormURL(query string) string {
 
 	return ""
 }
+
+//===========================================================================
+// Ranking Search Results
+//===========================================================================
+
+type SearchRank struct {
+	query string
+	limit int
+	items RankList
+	idset map[ulid.ULID]struct{}
+}
```

Comment:
Encapsulated all of the ranking and code including deduplication into a single struct to make it easier to test.